validator/index.js

```diff
@@ -25,50 +25,233 @@ const path = require('path');
 const program = require('commander');
 const vm = require('vm');
 const url = require('url');
+const util = require('util');
 
 /**
- * If the provided fileOrUrl start with 'http://' or 'https://', downloads
- * with the http or https module, otherwise reads the file with the file module.
- * Or, if the provided fileOrUrl is '-', reads from stdin.
- * Then, calls onSuccess with the data, or onFailure with the error message.
- * This method will check the HTTP status code as well and generate
- * brief error messages if it's not 200.
- * @param {!string} fileOrUrl
- * @param {!function(string)} onSuccess
- * @param {!function(string)} onFailure
+ * Convenience function to detect whether an argument is a ULR or
+ * (perhaps) a local file.
```

Comment:
"perhaps a local file", is that still correct? We're only checking for `/https?:\/\//`